deltacat/utils/pyarrow.py

```diff
@@ -476,10 +476,11 @@ def s3_file_to_parquet(
         content_type != ContentType.PARQUET.value
         or content_encoding != ContentEncoding.IDENTITY
     ):
-        raise ValidationError(
-            f"S3 file with content type: {content_type} and "
-            f"content encoding: {content_encoding} cannot be read"
-            "into pyarrow.parquet.ParquetFile"
+        raise ContentTypeValidationError(
```

Comment:
I also didn't see changes in `task_options.py` where Ray must be instructed to retry any `RetryableError`.

Reply:
For this feature, we have at least two-level of retryable errors: Ray task level and job level. DeltaCAT retryable error are meant for the job-level so that our driver can catch the error that shouldn't terminate the job.
For Ray task level retries, I feel like our current behavior is better by catch specific dependency error like Botocore throttling instead of casting to a DeltaCAT error. 
In addition, if we want to retry OOM Ray task, we should just catch like RayWorkerDiedError and retry based on this specific error. The Retry could be specific to each retry scenario instead of a general one.

Reply:
The task level retries won't work anymore because you are converting the underlying errors to `DeltaCatError`. So, we still need to handle that.

Reply:
That's why I didn't catch the RayWorkerDied, OutOfMemoryError, WorkerCrashedError, etc.

Reply:
You were catching all exceptions but not throwing any exceptions in its not RuntimeEnvError. Also it is confusing and violates SRP.

Reply:
I didn't see this comment addressed. Are you taking it up as a separate PR?

Reply:
Yes, all retry logic will be in second PR

Reply:
Ok..